api/v1alpha1/warehouse_types.go

```diff
@@ -267,6 +284,127 @@ type WarehouseStatus struct {
 	ObservedGeneration int64 `json:"observedGeneration,omitempty" protobuf:"varint,4,opt,name=observedGeneration"`
 	// LastFreight refers to the last Freight produced by this Warehouse
 	LastFreight *FreightReference `json:"lastFreight,omitempty" protobuf:"bytes,5,opt,name=lastFreight"`
+	// DiscoveredArtifacts holds the artifacts discovered by the Warehouse.
+	DiscoveredArtifacts *DiscoveredArtifacts `json:"discoveredArtifacts,omitempty" protobuf:"bytes,7,opt,name=discoveredArtifacts"`
+}
+
+// DiscoveredArtifacts holds the artifacts discovered by the Warehouse.
+type DiscoveredArtifacts struct {
+	// Commits holds the Git commits discovered by the Warehouse.
+	Commits []GitDiscoveryResult `json:"commits,omitempty" protobuf:"bytes,1,rep,name=commits"`
+
+	// Images holds the images discovered by the Warehouse.
+	Images []ImageDiscoveryResult `json:"images,omitempty" protobuf:"bytes,2,rep,name=images"`
+
+	// Charts holds the charts discovered by the Warehouse.
+	Charts []ChartDiscoveryResult `json:"charts,omitempty" protobuf:"bytes,3,rep,name=charts"`
+}
+
+// GitDiscoveryResult represents the result of a Git discovery operation for a
+// GitSubscription.
+type GitDiscoveryResult struct {
+	// RepoURL is the repository URL of the GitSubscription.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Pattern=`^https?://(\w+([\.-]\w+)*@)?\w+([\.-]\w+)*(:[\d]+)?(/.*)?$`
+	RepoURL string `json:"repoURL" protobuf:"bytes,1,opt,name=repoURL"`
```

Comment:
Question applies in several places. I'm curious what the value of the kubebuilder markers might be, if any, for things like this, which (admittedly early in my review) seem only to be system-generated and not user-generated.

Reply:
1. If the system is wrong due to a (presumably) human mistake, this is one way to figure this out soon.
2. It documents what can be expected of the value, which has value by itself.

Reply:
Works for me.